### PR TITLE
Add table aws_trusted_advisor_check_summary Closes #1831

### LINKED ISSUE #1831: Add table aws_trusted_advisor
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.299/service/support#Support.DescribeTrustedAdvisorCheckSummaries

Note: This table is required to check if AWS Trusted Advisor is enabled with zero checks in alarm state.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
I wasn't able to personally test the pull request because we lack a business, Enterprise On-Ramp, or Enterprise Support plan, which is required for making the `DescribeTrustedAdvisorCheckSummaries` API call. So the PR is in draft mode.

**Comment by ParthaI** [comment]
This PR is good to review now. It has been tested by the user(https://github.com/turbot/steampipe-plugin-aws/issues/1831#issuecomment-1829056632).

File: docs/tables/aws_trusted_advisor_check_summary.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  check_id,
+  category,
+  status
+from
+  aws_trusted_advisor_check_summary
+where
+  status = 'error';
+```
+
+### Get last 5days check summaries
```
Comment (madhushreeray30) [inline docs/tables/aws_trusted_advisor_check_summary.md:38]:
```suggestion
### Get check summaries for the last 5 days
```

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI please have a look at the comments, thanks!

File: docs/tables/aws_trusted_advisor_check_summary.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Table: aws_trusted_advisor_check_summary
```
Comment (khushboo9024) [inline docs/tables/aws_trusted_advisor_check_summary.md:1]: @ParthaI Please update the document in the latest format.

File: aws/table_aws_trusted_advisor_check_summary.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+func getTrustedAdvisorCheckSummary(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	checkId := ""
+
+	if h.Item != nil {
+		checkSummary := h.Item.(types.TrustedAdvisorCheckDescription)
+		checkId = *checkSummary.Id
+	} else if d.EqualsQualString("check_id") != "" {
+		checkId = d.EqualsQualString("check_id")
+	}
```
Comment (misraved) [inline aws/table_aws_trusted_advisor_check_summary.go:168]: Is the formatting correct?

File: aws/table_aws_trusted_advisor_check_summary.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+	if svc == nil {
+		return nil, nil
+	}
+
+	input := &support.DescribeTrustedAdvisorChecksInput{
+		Language: aws.String("en"),
+	}
+
+	// List call
+	result, err := svc.DescribeTrustedAdvisorChecks(ctx, input)
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply aws/table_aws_trusted_advisor_check_summary.go:139]: No, the [API](https://docs.aws.amazon.com/awssupport/latest/APIReference/API_DescribeTrustedAdvisorChecks.html) does not support pagination

File: aws/table_aws_trusted_advisor_check_summary.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_trusted_advisor_check_summary.listTrustedAdvisorCheckSummaries", "client_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		return nil, nil
+	}
+
+	input := &support.DescribeTrustedAdvisorChecksInput{
+		Language: aws.String("en"),
```
[the remark this answers is not in the document]
Comment (misraved) [reply aws/table_aws_trusted_advisor_check_summary.go:135]: @ParthaI I think we can update the table to use `language` as a required qual to best replicate the API behaviour.

**Comment by misraved** [comment]
@ParthaI could you please update the example queries to use the required `language` qual?

**Review by misraved — APPROVED** [review]

**Pull request merged**